lib/amazon_ssa_support/instance_metadata.rb

```diff
@@ -0,0 +1,44 @@
+require 'httpclient'
+
+module AmazonSsaSupport
+  class InstanceMetadata
+    
+    def initialize(version='latest')
+      @baseUrl    = 'http://169.254.169.254/'
+      @version    = version
+      @url      = "#{@baseUrl}#{@version}/"
+      @metadataUrl  = "#{@url}meta-data/"
+      @httpClient    = HTTPClient.new
+    end
+    
+    def version=(val)
+      @version = val
+      @url = "#{@baseUrl}#{@version}/"
+      @metadataUrl = "#{@url}meta-data/"
+      return val
+    end
+    
+    def versions
+      do_get(@baseUrl, "versions").split("\n")
+    end
+    
+    def metadata(path)
+      rv = do_get(@metadataUrl + path, "metadata")
+      return rv.split("\n") if rv.include?("\n")
+      return rv
```

Comment:
I'm assuming that the response from httpClient.get is a string in this comment.  This method is confusing because if my assumption is correct, then it returns an array if the response contains a newline, otherwise it returns a string.  Is this correct?  Also, no need for "return" in the last line.